futures-util/src/try_future/mod.rs

```diff
@@ -409,23 +444,26 @@ pub trait TryFutureExt: TryFuture {
     }
 */
 
-    /// Handle errors generated by this future by converting them into
-    /// `Self::Item`.
+    /// Unwraps this future's ouput, producing a future with this future's
+    /// [`Ok`](TryFuture::Ok) type as
+    /// [`Output`](std::future::Future::Output) type.
     ///
-    /// Because it can never produce an error, the returned `UnwrapOrElse` future can
-    /// conform to any specific `Error` type, including `Never`.
+    /// If this futures is resolved successfully, the returned future will
```

Comment:
Maybe "the original future" rather than "this future"? It's not totally clear which future is "this future" to me